spec/models/preneeds/form_fill_spec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.describe Preneeds::FormFill do
+  context 'with valid attributes' do
+    subject { described_class.new }
+
+    let(:params) { attributes_for :form_fill }
+
+    let(:attachment_types) do
+      Common::Collection.new(Preneeds::AttachmentType, data: attributes_for(:preneeds_attachment_type))
+    end
+
+    let(:branches_of_services) do
+      Common::Collection.new(Preneeds::BranchesOfService, data: attributes_for(:branches_of_service))
+    end
+
+    let(:cemeteries) do
+      Common::Collection.new(Preneeds::Cemetery, data: attributes_for(:cemetery))
+    end
+
+    let(:states) do
+      Common::Collection.new(Preneeds::State, data: attributes_for(:preneeds_state))
+    end
+
+    let(:discharge_types) do
+      Common::Collection.new(Preneeds::DischargeType, data: attributes_for(:discharge_type))
+    end
+
+    it 'populates attributes' do
+      allow_any_instance_of(Preneeds::Service).to receive(:expired?).and_return(true)
+      allow_any_instance_of(Preneeds::Service).to receive(:get_attachment_types).and_return(attachment_types)
+      allow_any_instance_of(Preneeds::Service).to receive(:get_branches_of_service).and_return(branches_of_services)
+      allow_any_instance_of(Preneeds::Service).to receive(:get_cemeteries).and_return(cemeteries)
+      allow_any_instance_of(Preneeds::Service).to receive(:get_states).and_return(states)
+      allow_any_instance_of(Preneeds::Service).to receive(:get_discharge_types).and_return(discharge_types)
```

Comment:
At first I did use :new, but it ended up being more work in the expect-spec part of the test because in the `expect(abc).to include(xyz)` part, there were some issues regarding the <=> operator. Not sure why, as I went with expediency by simply mocking up what is returned by the client. That solved the issue, and made the `expect(abc).to include(xyz)` much easier - I can spend some time today working on the change to `:new`